tools/export/iar/iar_definitions.json

```diff
@@ -0,0 +1,171 @@
+{
```

Comment:
These lines under the section "Steps to test or reproduce" are really useful.

As far as the new keys to `/hal/targets.json`, can those be added to the doc here: https://github.com/ARMmbed/mbed-os/blob/master/docs/mbed_targets.md